src/main/Main.hs

```diff
@@ -138,6 +140,16 @@ main = withInterpreterArgs stackProgName $ \args isInterpreter -> do
          "stack - The Haskell Tool Stack"
          ""
          (globalOpts False)
+         -- When there's an invalid argument
+         (Just $ \parseResult as -> do
+           mExternalExec <- Directory.findExecutable ("stack-" ++ head as)
+           case mExternalExec of
+             Just ex -> do
+               ph <- spawnProcess ex (tail as)
```

Comment:
It would be better if this replaces the current process instead of spawning a sub-process and just waiting for it to finish.  `Stack.Exec.exec` will do that, but it's another function that needs an EnvOverride (this time via a Config).  It's possible to create an EnvOverride using `System.Process.Read.mkEnvOverride`, and `exec` could be modified to take an EnvOverride as an argument instead of needing a Config.